server/enterprise.go

```diff
@@ -77,68 +78,68 @@ func (s *Server) getPRInfo(ctx context.Context, pr *model.PullRequest) (*EETrigg
 		return nil, fmt.Errorf("owner of server branch not found")
 	}
 
-	webappOwner, webappBranch, err := s.findWebappBranch(clientGitHub, ctx, pullRequest)
+	eeBranch, err := s.getBranchWithSameName(ctx, s.Config.Org, s.Config.EnterpriseReponame, pr.Ref)
 	if err != nil {
 		return nil, err
 	}
-	info := &EETriggerInfo{
-		BaseBranch:   *pullRequest.Base.Ref,
-		ServerOwner:  serverOwner,
-		ServerBranch: pr.Ref,
-		WebappOwner:  webappOwner,
-		WebappBranch: webappBranch,
+	if eeBranch == "" {
+		eeBranch = baseBranch
 	}
-	return info, nil
-}
-
-func (s *Server) findWebappBranch(client *github.Client, ctx context.Context, serverPR *github.PullRequest) (string, string, error) {
-	serverBranch := serverPR.GetHead().GetRef()
-	destinationServerBranch := serverPR.GetBase().GetRef()
 
-	owner, webappBranch, err := s.getWebappBranchWithSameName(client, ctx, serverPR)
+	webappOwner, webappBranch, err := s.getSameBranchInRepo(ctx, pr, s.Config.EnterpriseWebappReponame)
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
-
 	if webappBranch == "" {
-		mlog.Debug("Setting base ref of server as webapp branch", mlog.Int("pr", serverPR.GetNumber()), mlog.String("ref", serverBranch))
-		return s.Config.Org, destinationServerBranch, nil
+		webappOwner = s.Config.Org
+		webappBranch = baseBranch
 	}
 
-	return owner, webappBranch, nil
+	info = &EETriggerInfo{
+		BaseBranch:   baseBranch,
+		EEBranch:     eeBranch,
+		ServerOwner:  serverOwner,
+		ServerBranch: pr.Ref,
+		WebappOwner:  webappOwner,
+		WebappBranch: webappBranch,
+	}
+	return info, nil
 }
 
-func (s *Server) getWebappBranchWithSameName(client *github.Client, ctx context.Context, serverPR *github.PullRequest) (owner string, branch string, err error) {
-	prAuthor := serverPR.GetUser().GetLogin()
-	ref := serverPR.GetHead().GetRef()
-	forkBranch, r, err := client.Repositories.GetBranch(ctx, prAuthor, s.Config.EnterpriseWebappReponame, ref)
+func (s *Server) getBranchWithSameName(ctx context.Context, remote string, repo string, ref string) (branch string, err error) {
```

Comment:
I think we should pass the GitHub client here in order to avoid creating a new one every time.

I've seen that we're creating that client with the same parameter always so maybe we could move that to the server struct and use it from there and in case we need a new one could be created with a different token in another place

Reply:
That's the way to properly handle this! Nice, thank you.  I don't want to touch it everywhere in this PR, but I will slowly move it there and will do it for these instances now.

Reply:
Agree. Thank you!

Reply:
Maybe that's worth a HW in this repo? Community could help with the migration.

Reply:
I think that's a really good idea :)